source/gmsh/utilities.cc

```diff
@@ -100,11 +100,13 @@ namespace Gmsh
 
     if (base_name != prm.output_base_name)
       {
-        for (const auto &filename: {iges_file_name,geo_file_name,msh_file_name,log_file_name,warnings_file_name})
-        {
-          const auto ret_value = std::remove(filename.c_str());
-          AssertThrow(ret_value == 0, ExcMessage("Failed to remove "+filename));
-        }
+        const std::array<const std::string *, 5> filenames
+        {{&iges_file_name, &geo_file_name, &msh_file_name, &log_file_name, &warnings_file_name}};
+        for (const std::string *filename : filenames)
```

Comment:
This looks like a candidate for something someone might want to clean up to the original form sometime soon... As a follow-up (for mainline), would you mind leaving a one-sentence comment that explains why you didn't inline the list? Maybe it's also possible to make this shorter by writing
```
 auto filenames = {...};
  for (const std::string *filename : filenames)
```
The actual type of `filenames` is not important to us, and using `std::array` seems heavy-handed here. But then I have no idea whether that actually works around the compiler error.

Reply:
The compiler error is a bit silly: I can only reproduce it when I use C++11 with `-fconcepts` (which makes no sense). Its unlikely anyone else will hit it, but I'll post a clean up.

Reply:
The GCC concepts implementation has a lot of other weird bugs:

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=84979

Reply:
Sigh :-( But we won't be using concepts until at least 2022 or so, so at least we don't need this right now :-)